internal/pkg/uki/generate.go

```diff
@@ -206,40 +266,72 @@ func (builder *Builder) generateKernel() error {
 	return nil
 }
 
+type profileIndex struct {
+	Start int
+	End   int
+}
+
 func (builder *Builder) generatePCRSig() error {
-	sectionsData := xslices.ToMap(
-		xslices.Filter(builder.sections,
-			func(s section) bool {
-				return s.Measure
-			},
-		),
-		func(s section) (string, string) {
-			return s.Name, s.Path
-		})
+	profileIndexes := []profileIndex{}
```

Comment:
does it work with profile-less UKIs? (I don't see a test for it?)

Reply:
good point, let me add a test